yt_dlp/extractor/vidlii.py

```diff
@@ -45,10 +67,33 @@ def _real_extract(self, url):
 
         webpage = self._download_webpage(
             'https://www.vidlii.com/watch?v=%s' % video_id, video_id)
-
-        video_url = self._search_regex(
-            r'src\s*:\s*(["\'])(?P<url>(?:https?://)?(?:(?!\1).)+)\1', webpage,
-            'video url', group='url')
+        formats = []
+
+        def add_format(format_url, height=None):
+            try:
+                height = int(self._search_regex(r"(?P<h>\d\d\d).mp4",
+                                                format_url, "height",
+                                                group="h"))
+            except Exception:
+                height = 360
+            formats.append({
+                'url': format_url,
+                'format_id': '%dp' % height if height else None,
+                'height': height,
+            })
+
+        hdsrc = self._search_regex(
+            r'hdsrc\s*:\s*(["\'])(?P<url>(?:https?://)?(?:(?!\1).)+)\1',
+            webpage, 'video url', group='url')
+        try:
+            compat_urllib_request.urlopen(hdsrc)
+            add_format(hdsrc)
+        except Exception:
+            pass
+        add_format(self._search_regex(
+            r'src\s*:\s*(["\'])(?P<url>(?:https?://)?(?:(?!\1).)+)\1',
+            webpage, 'video url', group='url'))
```

Comment:
the two regexes are almost identical. This can be generalized

Reply:
ok, I changed it. should anything else be done?